### DISCUSSION TITLE
Merge two accounts into one repository (QCadjunct) and both accounts are owned by the  Peter.Heller@qc.cuny.edu email

### DISCUSSION THREAD

**Discussion by ph3ll3r** [community/community#133832] [GitHub Education]
### Hi everyone,

# My academic email is Peter.Heller@qc.cuny.edu.  

1. It is connected with ph3ll3r and QCadjunct. 
2. I want to merge ph3ll3r into QCadjunct which has been approved for the GItHub classroom and faculty benefits.  
3. Also, all of the ph3ll3r repositories are for exclusive for academic purposes and my learning for course implementation.

## I want one repository owned by QCadjunct and the  Peter.Heller@qc.cuny.edu email from Queens College.

# How can this be rectified?

**Comment by harisdev-netizen** [comment]
I understand you want to merge your GitHub accounts and reorganize your repositories. Let's break this down step-by-step:

1. Merging GitHub accounts: Unfortunately, GitHub doesn't have a direct "merge accounts" feature. However, we can achieve a similar result by transferring repositories and updating your primary email.
2. Transferring repositories: To move repositories from ph3ll3r to QCadjunct: a. Log in to the ph3ll3r account. b. For each repository:
- Go to the repository's settings.
- Scroll down to the "Danger Zone".
- Click "Transfer ownership".
- Enter the QCadjunct username as the new owner.
- Confirm the transfer.

3. Updating primary email: To associate [Peter.Heller@qc.cuny.edu](mailto:Peter.Heller@qc.cuny.edu) with QCadjunct: a. Log in to the QCadjunct account. b. Go to Settings > Emails. c. Add [Peter.Heller@qc.cuny.edu](mailto:Peter.Heller@qc.cuny.edu) as a new email. d. Verify the email. e. Set it as your primary email if desired.
4. Cleaning up: After transferring all repositories and updating emails: a. You can choose to delete the ph3ll3r account if you no longer need it. b. Update any necessary collaborator permissions on the transferred repositories.
5. GitHub Classroom and faculty benefits: Since QCadjunct is already approved for GitHub Classroom and faculty benefits, you're all set here. Just ensure you're using this account for all academic-related activities.
6. Single repository owned by QCadjunct: If you want to combine all your academic work into a single repository: a. Create a new repository under QCadjunct. b. Import or manually add contents from your other repositories. c. Organize the content using directories for different courses or topics. d. Update the README to explain the repository's structure and purpose.
7. git remote set-url origin https://github.com/QCadjunct/new-repo-name.git
`git remote set-url origin https://github.com/QCadjunct/new-repo-name.git`
8. Profile cleanup: Update your QCadjunct profile to reflect your academic status and link to your institution.

Remember to review GitHub's terms of service and your institution's policies regarding account ownership and content sharing.
Is there any specific part of this process you'd like me to elaborate on?

**Comment by samus-aran** [comment]
Hi @ph3ll3r  :wave:!

It’s not possible to transfer Global Campus benefits over from one GitHub account to another.

However, you are able to merge your accounts, but the account with the benefits would be the main account. You can add your non-academic email from the previous account to the account containing the academic email. If you desire you can then rename this account once merged. For more details check out out the [merging of your personal accounts](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts) documentation.

**Comment by Sivamani-18** [comment]
To merge your GitHub accounts and consolidate your repositories under a single account (QCadjunct) while ensuring your academic email (Peter.Heller@qc.cuny.edu) is associated with it, follow these steps:

1. **Add Email to QCadjunct Account**:
    - Log in to your QCadjunct GitHub account.
    - Go to **Settings** > **Emails**.
    - Add your academic email (Peter.Heller@qc.cuny.edu) to this account.
    - Verify the email address through the verification email sent by GitHub.

2. **Transfer Repositories from ph3ll3r to QCadjunct**:
    - Log in to your ph3ll3r GitHub account.
    - For each repository you want to transfer:
        - Go to the repository's main page on GitHub.
        - Click on **Settings**.
        - Scroll down to the **Danger Zone** section and click **Transfer**.
        - Enter the repository name and the new owner's GitHub username (QCadjunct).
        - Confirm the transfer.

3. **Set Up QCadjunct as the Main Account**:
    - Ensure all repositories are now under QCadjunct.
    - Update any links or references to the old repositories in your course materials, syllabi, and documentation to point to the new locations under QCadjunct.

4. **Update GitHub Classroom and Faculty Benefits**:
    - If your GitHub Classroom and faculty benefits are associated with QCadjunct, you don't need to do anything further.
    - If they are still associated with ph3ll3r, you may need to reapply or update the details to reflect QCadjunct. Check GitHub's [GitHub Education](https://education.github.com/) page for specific instructions.

5. **Close or Archive ph3ll3r Account** (Optional):
    - If you no longer need the ph3ll3r account, you can close it. However, consider keeping it for personal projects if desired.
    - To close the account, go to **Settings** > **Account** and scroll down to the **Delete Account** section.

By following these steps, you will have merged your repositories into the QCadjunct account and ensured your academic email is associated with this account. This will streamline your academic work and take advantage of GitHub's educational benefits.
